src/query/service/tests/it/storages/fuse/operations/mutation/recluster_mutator.rs

```diff
@@ -388,11 +388,10 @@ async fn generage_segments(
     }
 
     let threads_nums = ctx.get_settings().get_max_threads()? as usize;
-    let permit_nums = ctx.get_settings().get_max_storage_io_requests()? as usize;
     execute_futures_in_parallel(
```

Comment:
Here no need to use `execute_futures_in_parallel`?

Reply:
Ping @dantengsky  for better suggestion

Reply:
yeah, @JackTan25 noticed that exceute_futures_in_parallel could be replaced with `try_join_all`,

but need to confirm further if in all the cases `permit_nums` eq `threads_nums` is reasonable.

Reply:
We can wrapper a new method for it? Like:
```
pub async fn execute_futures_parallel<Fut>(
    futures: impl IntoIterator<Item = Fut>,
    thread_nums: usize,
    thread_name: String,
) -> Result<Vec<Fut::Output>>
```

And change the old one to:
```
pub async fn execute_futures_parallel_with_permits<Fut>(
    futures: impl IntoIterator<Item = Fut>,
    thread_nums: usize,
    permit_nums: usize,
    thread_name: String,
) -> Result<Vec<Fut::Output>>
```

Reply:
@zhyass Any concerns?  shall we let the `permit_nums` eq `scale_factor * threads_nums`, where `scale_factor = 4`, in this case (reclustering/compaction) for example?

Reply:
> @zhyass Any concerns? shall we let the `permit_nums` eq `scale_factor * threads_nums`, where `scale_factor = 4`, in this case (reclustering/compaction) for example?

Set `scale_factor * threads_nums`, if use `execute_futures_in_parallel`. Otherwise use `threads_nums` directly.